presto-release-tools/src/test/java/com/facebook/presto/release/tasks/TestGenerateReleaseNotesTask.java

```diff
@@ -169,14 +168,14 @@ private static PullRequest loadPullRequest(String title, Person author, Person m
 
     private static List<Commit> createCommits(String title, Person author, Person mergedBy, boolean associateWithPullRequest, int commitCount)
     {
-        List<PullRequest> associatedPullRequests = associateWithPullRequest ? ImmutableList.of(loadPullRequest(title, author, mergedBy)) : ImmutableList.of();
+        int id = associateWithPullRequest ? PullRequestId.getAndIncrement() : 0;
```

Comment:
Why not keep the list construction here and increment the ID inline?

Reply:
This method returns `List<Commit>` all share same pull request. 
Before this refactor, one instance of pull request will be generated and passed to all commits. That was the reason this test case was not able to catch the duplicate bug.
With this refactor we pass the same `id` to `loadPullRequest` each time to get different instances of pull request objects that are equal by value (this is what happens in real cases when we call github api to get commits).

Note: The code change in this commit will pass the test even w/o this refactor. I was wondering why unit test was not able to catch the issue of duplicate messages in the first place and fixed the test itself here.